cmd/src/gateway_benchmark.go

```diff
@@ -0,0 +1,258 @@
+		requestCount = flagSet.Int("requests", defaultRequestCount, "Number of requests to make per endpoint")
+	)
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		if len(flagSet.Args()) != 0 {
+			return cmderrors.Usage("additional arguments not allowed")
+		}
+
+		// Create HTTP client with TLS skip verify
+		client := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			},
+		}
+
+		endpoints := map[string]string{
+			"HTTP":                fmt.Sprintf("%s/cody-gateway-call-http", cfg.Endpoint),
+			"HTTP then WebSocket": fmt.Sprintf("%s/cody-gateway-call-http-then-websocket", cfg.Endpoint),
+		}
+
+		fmt.Printf("Starting benchmark with %d requests per endpoint...\n", *requestCount)
+
+		var results []endpointResult
+
+		for name, url := range endpoints {
+			durations := make([]time.Duration, 0, *requestCount)
+			fmt.Printf("\nTesting %s...", name)
+
+			for i := 0; i < *requestCount; i++ {
+				duration := benchmarkEndpoint(client, url)
+				if duration > 0 {
+					durations = append(durations, duration)
+				}
+				fmt.Printf("\rTesting %s: %d/%d", name, i+1, *requestCount)
+			}
+			fmt.Println()
+
+			avg, p5, p95, median, total := calculateStats(durations)
+
+			results = append(results, endpointResult{
+				name:       name,
+				avg:        avg,
+				median:     median,
+				p5:         p5,
+				p95:        p95,
+				total:      total,
+				successful: len(durations),
+			})
+		}
+
+		printResults(results, requestCount)
+		return nil
+	}
+
+	gatewayCommands = append(gatewayCommands, &command{
+		flagSet: flagSet,
+		aliases: []string{},
+		handler: handler,
+		usageFunc: func() {
+			_, err := fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src gateway %s':\n", flagSet.Name())
+			if err != nil {
+				return
+			}
+			flagSet.PrintDefaults()
+			fmt.Println(usage)
+		},
+	})
+}
+
+type endpointResult struct {
+	name       string
+	avg        time.Duration
+	median     time.Duration
+	p5         time.Duration
+	p95        time.Duration
+	total      time.Duration
+	successful int
+}
+
+func benchmarkEndpoint(client *http.Client, url string) time.Duration {
+	start := time.Now()
+	resp, err := client.Get(url)
+	if err != nil {
+		fmt.Printf("Error calling %s: %v\n", url, err)
+		return 0
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Printf("Error closing response body: %v\n", err)
+		}
+	}(resp.Body)
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("Error reading response body: %v\n", err)
+		return 0
+	}
+	fmt.Printf("Response from %s: %s\n", url, body)
+
+	return time.Since(start)
+}
+
+func calculateStats(durations []time.Duration) (time.Duration, time.Duration, time.Duration, time.Duration, time.Duration) {
+	if len(durations) == 0 {
+		return 0, 0, 0, 0, 0
+	}
+
+	sort.Slice(durations, func(i, j int) bool {
+		return durations[i] < durations[j]
```

Comment:
adding a note here for me to triple check we got the order right here; just don't want any chance we've got p95 and p5 inverted.

Actually; could you add p80 and p75 as well? that'd make it clear and might be useful to collect anyway.